spec/features/course/assessment/question/duplication_spec.rb

```diff
@@ -6,42 +6,101 @@
+
+          within all('section', text: question.title).first do
+            click_button 'Duplicate'
+          end
+
+          find('li', text: first_destination_assessment.title).click
+          expect_toastify('Your question has been duplicated.')
+
+          wait_for_page
 
-      scenario 'I can duplicate a question from one assessment to another' do
-        destination_assessment
-        visit course_assessment_path(course, source_assessment)
+          expect(first_destination_assessment.questions.reload.count).to eq(1)
 
-        expect(destination_assessment.questions.count).to be(1)
+          duplicated_question = first_destination_assessment.questions.last
+          expect(duplicated_question.title).to eq(question.title)
+          expect(duplicated_question.question_assessments.first.skills.first.id).
+            to eq(question.question_assessments.first.skills.first.id)
+        end
+      end
+
+      context 'upon duplicating deprecated programming language question' do
+        let!(:second_destination_assessment) { create(:assessment, course: course) }
+        let!(:language) { programming_question.specific.language }
 
-        within all('section', text: source_assessment.questions.first.title).first do
-          click_button 'Duplicate'
+        # Disable the language before the test, and enable it after the test
+        # Direct SQL is used to avoid the readonly limitations
+        before do
+          ActiveRecord::Base.connection.execute(
+            "UPDATE polyglot_languages SET enabled = false WHERE id = #{language.id}"
+          )
+          programming_question.reload
         end
 
-        wait_for_job
+        after do
+          ActiveRecord::Base.connection.execute(
+            "UPDATE polyglot_languages SET enabled = true WHERE id = #{language.id}"
+          )
+        end
+
+        scenario 'I can also duplicate that question from one assessment to another' do
+          visit course_assessment_path(course, second_source_assessment)
+
+          expect(second_destination_assessment.questions.count).to be(0)
 
-        expect do
-          find('li', text: destination_assessment.title).click
+          within all('section', text: programming_question.title).first do
+            click_button 'Duplicate'
+          end
+
+          find('li', text: second_destination_assessment.title).click
           expect_toastify('Your question has been duplicated.')
-        end.to change { destination_assessment.questions.count }.by(1)
 
-        original_mcq_question = source_assessment.questions.last
-        duplicated_mcq_question = destination_assessment.questions.last
-        expect(duplicated_mcq_question.title).to eq(original_mcq_question.title)
-        expect(duplicated_mcq_question.question_assessments.first.skills.first.id).
-          to eq(original_mcq_question.question_assessments.first.skills.first.id)
+          wait_for_page
```

Comment:
`wait_for_page` is unnecessary, don't include it unless it's actually needed since it will cause our tests duration to lengthen.

`expect(page).to have_text` already has built-in maximum wait time.

Reply:
done